dog_classifier/net/train.py

```diff
@@ -40,40 +60,58 @@ def trainNN(training_parameters):
     early_stopping_patience = training_parameters['early_stopping_patience']
     early_stopping_delta = training_parameters['early_stopping_delta']
 
-    df_train = pd.read_csv(path_to_labels + 'train_labels.csv')
-    df_val = pd.read_csv(path_to_labels + 'val_labels.csv')
-
-    trainDataloader = DataGenerator(df_train, encoder_model, batch_size=bs_size,
-                                    n_classes=n_classes)
-    valDataloader = DataGenerator(df_val, encoder_model, batch_size=bs_size,
-                                  n_classes=n_classes)
-
     model = get_model(training_parameters['architecture'])
     # Set the leranrning rate of adam optimizer
     Adam(training_parameters['learning_rate'])
 
+    df_train = pd.read_csv(path_to_labels + 'train_labels.csv')
+    df_val = pd.read_csv(path_to_labels + 'val_labels.csv')
+    with K.tf.device('/cpu:0'):
```

Comment:
Warum jetzt den Dataloader in der CPU? Geht das massiv schneller?

Reply:
Um die GPU zu entlasen, damit die sich auf die BackPropargation konzentrieren kann. 
Es könnte sein, dass er aber schon per default auf der CPU läuft.